packages/backend-plugin-api/report.api.md

```diff
@@ -24,6 +25,80 @@ import { Readable } from 'stream';
 import { Request as Request_2 } from 'express';
 import { Response as Response_2 } from 'express';
 
+// @public
+export type AuditorEvent = [
+  message: string,
+  meta: {
+    actor: AuditorEventActorDetails;
+    eventName: string;
+    stage: string;
+    meta?: JsonObject;
+    request?: AuditorEventRequest;
+  } & AuditorEventStatus,
+];
+
+// @public (undocumented)
+export type AuditorEventActorDetails = {
+  actorId?: string;
+  ip?: string;
+  hostname?: string;
+  userAgent?: string;
+};
+
+// @public
+export type AuditorEventArgs<T extends JsonObject> = {
+  message: string;
+  eventName: string;
+  stage: string;
+  request?: Request_2;
+  actorId?: string;
+  meta?: T;
+} & AuditorEventStatus;
+
+// @public
+export type AuditorEventFailureStatus<E = ErrorLike> = {
+  status: 'failed';
+  errors: E[];
+};
+
+// @public (undocumented)
+export type AuditorEventRequest = {
+  url: string;
+  method: string;
+};
+
+// @public (undocumented)
+export type AuditorEventStatus =
+  | AuditorEventSuccessStatus
+  | AuditorEventFailureStatus
+  | AuditorEventUnknownStatus;
+
+// @public
+export type AuditorEventSuccessStatus = {
+  status: 'succeeded';
+};
+
+// @public
+export type AuditorEventUnknownStatus = {
+  status: 'unknown';
+};
+
+// @public
+export interface AuditorService {
+  // (undocumented)
+  child(meta: JsonObject): AuditorService;
+  // (undocumented)
+  debug<T extends JsonObject>(args: AuditorEventArgs<T>): Promise<void>;
+  // (undocumented)
+  error<T extends JsonObject>(args: AuditorEventArgs<T>): Promise<void>;
+  // (undocumented)
+  getActorId(request: Request_2): Promise<string | undefined>;
```

Comment:
I'm having a hard time seeing how this is intended to be used and why it needs to be a separate method, could you perhaps add something around that to https://github.com/backstage/backstage/tree/master/beps/0011-event-auditor#design-details?

Reply:
Before I make any additions to the BEP, it would be helpful to get a response to this [thread](https://github.com/backstage/backstage/pull/26372#discussion_r1747119584). In short, the change was made to conserve compute resources.

Reply:
As discussed in the framework SIG, let's move this just live in the implementation instead